src/openassetio-python/cmodule/src/hostApi/ManagerBinding.cpp

```diff
@@ -181,50 +181,52 @@ void registerManager(const py::module& mod) {
           py::arg("context").none(false), py::arg("successCallback"), py::arg("errorCallback"),
           py::arg("resultTraitSet") = trait::TraitSet{})
       .def("preflight",
-           static_cast<void (Manager::*)(
-               const EntityReferences&, const trait::TraitSet&, const ContextConstPtr&,
-               const Manager::PreflightSuccessCallback&,
-               const Manager::BatchElementErrorCallback&)>(&Manager::preflight),
-           py::arg("entityReferences"), py::arg("traitSet"), py::arg("context").none(false),
-           py::arg("successCallback"), py::arg("errorCallback"))
+           py::overload_cast<const EntityReferences&, const trait::TraitsDatas&,
+                             const ContextConstPtr&, const Manager::PreflightSuccessCallback&,
+                             const Manager::BatchElementErrorCallback&>(&Manager::preflight),
+           py::arg("entityReferences"), py::arg("entityTraitsDatas"),
+           py::arg("context").none(false), py::arg("successCallback"), py::arg("errorCallback"))
       .def("preflight",
-           static_cast<EntityReference (Manager::*)(
```

Comment:
[Question] Is the change from `static_cast` to `py::overload_cast` a neccesary change or do you just prefer the style? I guess I wouldn't have expected a neccesary change here changing one arg type to another.